eng/Directory.Build.Common.props

```diff
@@ -106,6 +106,7 @@
     <GenerateAPIListing Condition="'$(GenerateAPIListing)' == '' AND '$(IsShippingClientLibrary)' == 'true'">true</GenerateAPIListing>
     <EnableSourceLink Condition="'$(EnableSourceLink)' == ''">true</EnableSourceLink>
     <DefineConstants Condition="'$(BuildSnippets)' == 'true'">$(DefineConstants);SNIPPET</DefineConstants>
+    <ApiCompatBaselineTargetFramework Condition="'$(TargetFramework)' == 'net8.0'">netstandard2.0</ApiCompatBaselineTargetFramework>
```

Comment:
Should we extract `net8.0` into a Build Prop variable so that we can change it in one place as we go through LTS targets over time?

Reply:
For this one it's temporary (tracking by https://github.com/Azure/azure-sdk-for-net/issues/46892), once enough libraries are released with net8.0 targets then this baseline can be moved from the common file to individual csproj files for SDKs that haven't released.